src/duplicate-results/sub-components/use-set-height-after-request.ts

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useRef, useState } from 'react';
+import { useAppSelector } from '../../app/hooks';
+import { selectDuplicateSearchTerm } from '../../duplicate-search/duplicate-search-slice';
+import { selectDuplicateResultsRequestStatus } from '../duplicate-results-slice';
+
+export function useSetHeightAfterRequest() {
+	const searchTerm = useAppSelector( selectDuplicateSearchTerm );
+	const resultsRequestStatus = useAppSelector( selectDuplicateResultsRequestStatus );
+	// This ref and corresponding useEffect hook are used to preserve the height of the
+	// results container between searches. This keeps the UI from jumping around while searching.
+	const resultsContainerContentRef = useRef< HTMLDivElement >( null );
+	const [ resultsContainerContentHeightPx, setResultsContainerContentHeightPx ] = useState<
+		number | undefined
+	>( undefined );
+
+	useEffect( () => {
+		if (
+			resultsRequestStatus === 'fulfilled' ||
+			resultsRequestStatus === 'error' ||
+			searchTerm === ''
```

Comment:
**suggestion (non-blocking):** If the `resultsRequestsStatus` is never `fulfilled` or `pending` when the `searchTerm` is empty, we can delete this condition.

Reply:
Oh right! Good catch! Specifically, we shouldn't need the `searchTerm` in this check -- because in the case when the search term is `empty`, the request will always be what it was last (which isn't pending).  So we should be able to just check for the request status, and then include both in the dependency array!